src/lib/services/feature-toggle-service.ts

```diff
@@ -1967,7 +1968,26 @@ class FeatureToggleService {
     }
 
     async updatePotentiallyStaleFeatures(): Promise<void> {
-        await this.featureToggleStore.updatePotentiallyStaleFeatures();
+        const potentiallyStaleFeatures =
+            await this.featureToggleStore.updatePotentiallyStaleFeatures();
+        if (this.flagResolver.isEnabled('emitPotentiallyStaleEvents')) {
+            if (potentiallyStaleFeatures.length) {
```

Comment:
We need to check that there are actual events in here, because otherwise the `batchStore` methods throws an error when it fails to insert the empty event data when there are no changes.